### PR TITLE
Update GraphCast for model standards

### PR DESCRIPTION
**Pull request by pzharrington**
# PhysicsNeMo Pull Request

## Description
Updates GraphCast to comply with model standards, along with misc fixes:

- Detailed shape documentation and annotation in the model itself
- Removal of deprecated `multimesh_level` option
- Moving recipe-specific items to the example
- Updating example to use torch fused AdamW instead of apex
- Update tests to drop old `dgl` reference

Tests are passing; transformer_engine dependency still needs to be worked out.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/weather/graphcast/conf/config.yaml [hunk 1]
```diff
@@ -61,8 +61,6 @@ cugraphops_encoder: false   # If true, uses cugraphops backend for the encoder.
 cugraphops_processor: false # If true, uses cugraphops backend for the processor.
 cugraphops_decoder: false   # If true, uses cugraphops backend for the decoder.
 recompute_activation: true  # If true, recomputes activation in backward to save memory. Currently, only SiLU is supported.
-use_apex: true              # If true, uses Apex for fused Adam optimizer, typically resulting in 10-20% faster training iterations.
```
Comment (mnabian) [inline examples/weather/graphcast/conf/config.yaml:64]: I'm curious, why are you removing support for apex?

File: physicsnemo/metrics/climate/graphcast_loss.py
Comment (mnabian) [inline physicsnemo/metrics/climate/graphcast_loss.py:1]: Do you think there is a way to keep this in the source code? This way we can also keep the test for GraphCast loss functions. For example, I see the healpix loss is in `metrics/climate`: https://github.com/NVIDIA/physicsnemo/blob/main/physicsnemo/metrics/climate/healpix_loss.py

**Review by mnabian — APPROVED** [review]
LGTM! Left a couple of comments.

File: examples/weather/graphcast/conf/config.yaml (hunk 1 above)
Comment (pzharrington) [reply examples/weather/graphcast/conf/config.yaml:64]: I thought it was upstreamed into core PyT via the fused option, in which case it's kinda redundant and nice to eliminate an extra dep. Would you prefer we keep apex here?

File: physicsnemo/metrics/climate/graphcast_loss.py
Comment (pzharrington) [reply physicsnemo/metrics/climate/graphcast_loss.py:1]: Migrated it into `metrics/climate/graphcast_loss.py` and put the test back in.

File: examples/weather/graphcast/conf/config.yaml (hunk 1 above)
Comment (mnabian) [reply examples/weather/graphcast/conf/config.yaml:64]: Oh yeah you are right. I remember previously there was a discrepancy between the Apex version and the PyT upstreamed version, which seems to be fixed: https://github.com/pytorch/pytorch/issues/96755
All good!

**Review by dallasfoster — APPROVED** [review]
